### PR TITLE
Allow EC keys to have a smaller bitsize

### PR DESCRIPTION
**Pull request by firesock**
Should prevent EC keys from showing up as fubar'd.

### CODE REVIEW

File: analyze.py [hunk 1]
```diff
@@ -43,14 +43,21 @@ def has_good_pfs(pfs, target_dh, target_ecc, must_match=False):
...
     has_wrong_pubkey = False
+    has_wrong_ec_pubkey = False
     has_bad_sig = False
     has_untrust_cert = False
     has_wrong_pfs = False
+
     for conn in results['ciphersuite']:
         logging.debug('testing connection %s' % conn)
+        pubkey_bits = int(conn['pubkey'][0])
+        ec_kex = conn['cipher'].startswith('ECDHE-')
```
Comment (tomato42) [inline analyze.py]: ` or conn['cipher'].startswith('ECDH-')` but I don't remember if then the pubkey_bits is the correct variable to look at

File: analyze.py [hunk 2]
```diff
@@ -43,14 +43,21 @@ def has_good_pfs(pfs, target_dh, target_ecc, must_match=False):
 def is_fubar(results):
     logging.debug('entering fubar evaluation')
     lvl = 'fubar'
+    min_ec_size = min(old["ecdh_param_size"], inter["ecdh_param_size"], modern["ecdh_param_size"])
```
Comment (tomato42) [inline analyze.py]: I think we can hardcode it, just like for rsa

File: analyze.py (hunk 1 above)
Comment (firesock) [inline analyze.py]: Good point. I've added a regex check to catch ciphers listed here https://www.openssl.org/docs/manmaster/apps/ciphers.html and a quick personal test shows that pubkey_bits is still correct for ECDH suites.

File: analyze.py (hunk 2 above)
Comment (firesock) [inline analyze.py]: I've gone with the minimum Mozilla size, 160, for now.

File: analyze.py [hunk 3]
```diff
@@ -70,7 +69,7 @@ def is_fubar(results):
             has_wrong_pubkey = True
             logging.debug(conn['pubkey'][0] + ' is a fubar pubkey size')
             fubar = True
-        if ec_kex and pubkey_bits < min_ec_size:
+        if ec_kex and pubkey_bits < 160:
```
Comment (tomato42) [inline analyze.py]: that should be 256: https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility

Comment (firesock) [inline analyze.py]: Hm, that contradicts the JSON versions:

https://statics.tls.security.mozilla.org/server-side-tls-conf-4.0.json 
https://statics.tls.security.mozilla.org/server-side-tls-conf.json

which still list 160 for old. Not sure which is correct here, and I don't have a MozillaWiki account to leave a comment on the discussion page.

Comment (tomato42) [inline analyze.py]: There are no clients that would support 160 bit curves but not the 256 bit curves. Either they support no curves, just P-256 and P-384 or P-256, P-384 and P-521 or all curves. So even for the legacy config there's no need to reduce the size of parameters, in fact, using 256 bit curves there should be mandatory to maximize interoperability.

@jvehent can you fix the json's?

Comment (firesock) [inline analyze.py]: That makes sense, I've pushed another commit. Just noticed that the has_good_pfs check in is_fubar still uses 160, but I've left that alone on the grounds that the DH parameter check is also only checking 1024, so not quite sure whether this is intentional.

**Pull request merged**